docs/source/federated-types/interfaces.mdx

```diff
@@ -0,0 +1,315 @@
+
+### Usage rules
+
+#### The `interface` definition
+
+Let's say Subgraph A defines the `MyInterface` type as an entity interface so that _other_ subgraphs can add fields to it:
+
+```graphql title="Subgraph A"
+interface MyInterface @key(fields: "id") {
+  id: ID!
+  originalField: String!
+}
+
+type MyObject implements MyInterface @key(fields: "id") {
+  id: ID!
+  originalField: String!
+}
+```
+
+**In this case:**
+
+- Subgraph A **must** include at least one `@key` directive in its `MyInterface` definition.
+    - It **may** include multiple `@key`s.
+- Subgraph A **must** define _every_ entity type in your _entire supergraph_ that implements `MyInterface`.
+    - Certain other subgraphs can _also_ define these entities, but Subgraph A **must** define **all** of them.
+    - You can think of a subgraph that defines an entity interface as also  _owning_ every entity that implements that interface.
+- Subgraph A **must** be able to uniquely identify _any_ instance of _any_ entity that implements `MyInterface`, using _only_ the `@key` fields defined by `MyInterface`.
+    - This means that if both `EntityA` and `EntityB` implement `MyInterface`, there must be _no_  exact `@key` field duplicates, even among different entities.
+    - This is required to support [resolving the interface](#required-resolvers) in Subgraph A.
+- Every entity that implements `MyInterface` **must** include **all** `@key`s from the `MyInterface` definition.
+    - These entities can _optionally_ define _additional_ `@key`s as needed.
+
+#### `@interfaceObject` definitions
+
+Let's say Subgraph B applies `@interfaceObject` to an object type named `MyInterface`:
+
+```graphql title="Subgraph B"
+type MyInterface @key(fields: "id") @interfaceObject {
+  id: ID!
+  addedField: Int!
+}
+```
+
+**In this case:**
+
+- At least one _other_ subgraph **must** define an _interface_ type named `MyInterface` with the `@key` directive applied to it (e.g., [Subgraph A above](#the-interface-definition))
+
+    ```graphql title="Subgraph A"
+    interface MyInterface @key(fields: "id") {
+      id: ID!
+      originalField: String!
+    }
+    ```
+
+- _Every_ subgraph that defines `MyInterface` as an object type **must**:
+    - Apply `@interfaceObject` to its definition
+    - Include the exact same `@key`(s) as the _interface_ type's definition
+- Subgraph B **must not** _also_ define `MyInterface` as an _interface_ type.
+- Subgraph B **must not** define _any_ entity that _implements_ `MyInterface`.
+    - If a subgraph contributes entity fields via `@interfaceObject`, it "gives up" the ability to contribute fields to any _individual_ entity that implements that interface.
+    - This is a current limitation of entity interfaces.
+
+
+- `@shareable` checks/rules on fields of   `Product` are performed “as if” the fields provided by `Product` in `reviews` were provided by all the types implementing `Product` in the supergraph.
```

Comment:
```suggestion
- `@shareable` checks/rules on fields of `Product` are performed “as if” the fields provided by `Product` in `reviews` were provided by all the types implementing `Product` in the supergraph.
```

Reply:
I'm not sure we are talking about the product type in this file other than this line

Reply:
ahh good catch this is a line from the previous draft that I forgot to edit 😬

Reply:
Going to cut this line for now, and we'll confirm the implications here w/ @pcmanus for a followup edit